flask_chat/views.py

```diff
@@ -0,0 +1,414 @@
+
+    log_request()
+    response = redirect("/login")
+    unset_jwt_cookies(response)
+
+    logger.info("User logged out")
+
+    return response
+
+
+@views_bp.route("/myprofile", methods=["GET", "POST"])
+@jwt_required()
+def profile():
+    """Display profile page"""
+
+    log_request()
+
+    user_id = get_jwt_identity()
+
+    # Retrieve info from database about user
+    user_data = user_service.get_user_by_id(user_id)
+    if not user_data:
+        logger.error("Unable to load user info.")
+        return render_template(
+            "error.html",
+            message="Unable to process info about your identity. Pleasy try later.",
+        )
+
+    user_data = user_data.to_json()
+    current_user = user_data.get("username")
+    logger.debug("Current user: %s", current_user)
+
+    # Get csrf token to put in rendered form
+    csrf_token = request.cookies.get("csrf_access_token")
+
+    # Edit profile form handle
+    form = EditProfileForm(request.form)
+
+    if request.method != "POST":
+
+        # Placeholder values from database to display in profile
+        for field in form:
+            if field.name == "csrf_token":
+                logger.debug("CSRF token is set")
+                field.data = csrf_token
+            elif user_data.get(field.name):
+                field.data = user_data.get(field.name)
+            # If the field from user's document is empty
+            else:
+                field.data = ""
+
+        return render_template(
+            "profile.html",
+            form=form,
+            current_user=current_user,
+            csrf_token=csrf_token,
+            data=user_data,
+            web_name=WEBSITE_NAME,
+        )
+
+    # Check if request contains data about file uploaded
+    if request.files["file"].read() != b"":
+        content = request.files["file"]
+        content.seek(0)
+        content_clean = content.read()
+        content.seek(0)
+
+        # Check if received file's content is empty or if file was not chosen
+        if content_clean == b"":
+            flash("Empty upload file provided.")
+            return redirect(request.url)
+
+        if not verify_image(content_clean):
+            flash("The file is inappropriate or corrupted. Try again")
+            return redirect(request.url)
+
+        # Generate new random name for file that will be stored
+        picture_name = random_strings_generator()
+
+        # Saving file to it's defined path
+        with open(
+            os.path.join(PROFILE_PICTURE_STORAGE_PATH, picture_name), "wb"
+        ) as file:
+            content.save(file)
+
+        # remove old picture to avoid trashing
+        if user_data.get("profile_picture") is not None:
+            os.remove(
+                os.path.join(PROFILE_PICTURE_STORAGE_PATH, user_data.get("profile_picture"))
+            )
+            logger.debug("previous picture is removed")
+
+        # Update profile picture filename in database
```

Comment:
Move this code up to avoid situation, when image deleted, but db update failed

Reply:
Fixed.